### PR TITLE
More manager reloading cleanups

### PR DESCRIPTION
**Pull request by keszybz**
No functional change, just trying to make the code nicer.

Follow-up for #10343.

### CODE REVIEW

File: src/core/manager.c [hunk 1]
```diff
@@ -3083,7 +3094,7 @@ int manager_serialize(
         assert(f);
         assert(fds);
 
-        m->n_reloading++;
+        _cleanup_(manager_reloading_stopp) Manager *reloading = manager_reloading_start(m);
```
Comment (poettering) [inline src/core/manager.c:3099]: hmm, still not a fan of mixing variable declarations with function calls in the same line

File: src/core/manager.c [hunk 2]
```diff
@@ -1653,66 +1653,68 @@ int manager_startup(Manager *m, FILE *serialization, FDSet *fds) {
 
         manager_build_unit_path_cache(m);
 
-        /* If we will deserialize make sure that during enumeration this is already known, so we increase the counter
-         * here already */
-        if (serialization)
-                m->n_reloading++;
+        {
+                /* This block is done (optionally) done with the reloading counter bumped */
```
Comment (dnicolodi) [inline src/core/manager.c]: `done` is repeated

File: src/core/manager.c [hunk 3]
```diff
@@ -3112,11 +3112,9 @@ int manager_serialize(
                         continue;
 
                 t = manager_timestamp_to_string(q);
-                {
-                        char field[strlen(t) + STRLEN("-timestamp") + 1];
-                        strcpy(stpcpy(field, t), "-timestamp");
-                        dual_timestamp_serialize(f, field, m->timestamps + q);
-                }
+                char field[strlen(t) + STRLEN("-timestamp") + 1];
```
Comment (dnicolodi) [inline src/core/manager.c]: I think this is the only place in the codebase where a VLA is used. I think `_cleanup_free_ char *field; field = strjoin(t, "-timestamp")` or the equivalent with `strjoina()` would be better.

Comment (poettering) [reply src/core/manager.c]: we use VLAs quite frequently actually. this is far from the only case.

**Comment by yuwata** [comment]
```
Failed to reload daemon: Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms)
Failed to reload daemon: Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms)
Failed to retrieve unit state: Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms)
Failed to restart udev.service: Failed to activate service 'org.freedesktop.systemd1': timed out (service_start_timeout=25000ms)
See system logs and 'systemctl status udev.service' for details.
```

File: src/core/manager.c (hunk 3 above)
Comment (dnicolodi) [reply src/core/manager.c]: Sorry, I thought that the `STRLEN` macro was introduced to avoid variable length arrays.

File: src/core/manager.c [hunk 4]
```diff
@@ -3568,14 +3540,6 @@ int manager_reload(Manager *m) {
 
         m->send_reloading_done = true;
         return 0;
```
Comment (yuwata) [inline src/core/manager.c:3543]: Missing `reloading = NULL;`. This function decreases `m->n_reloading` only when the above operation fails.

File: src/core/manager.c [hunk 5]
```diff
@@ -3491,19 +3467,15 @@ int manager_reload(Manager *m) {
...
+                return log_error_errno(r, "Failed to serialize manager: %m");
 
-        if (fseeko(f, 0, SEEK_SET) < 0) {
-                r = log_error_errno(errno, "Failed to seek to beginning of serialization: %m");
-                goto fail;
-        }
+        if (fseeko(f, 0, SEEK_SET) < 0)
+                return log_error_errno(errno, "Failed to seek to beginning of serialization: %m");
 
         /* 💀 This is the point of no return, from here on there is no way back. 💀 */
```
Comment (yuwata) [inline src/core/manager.c:3480]: It may be better to insert `reloading = NULL;` here.

Comment (keszybz) [reply src/core/manager.c:3480]: @yuwata: thanks!

Updated with the suggested changes, and the VLA changed to strjoina (it's just nicer in this case). No other changes.

**Comment by yuwata** [comment]
Lgtm.

**Pull request merged**